Sources/Controllers/DashboardOnMap/MapSettings/OAMapSettingsMainScreen.mm

```diff
@@ -402,57 +393,160 @@ - (NSString *) getPOIDescription
+    else if ([key isEqualToString:@"layer_amenity_label"])
+        return [_settings.mapSettingShowPoiLabel get];
+    else if ([key isEqualToString:@"wikipedia_layer"])
+        return _app.data.wikipedia;
+    else if ([key isEqualToString:@"osm_edits_offline_layer"])
+        return [_settings.mapSettingShowOfflineEdits get];
+    else if ([key isEqualToString:@"osm_notes_online_layer"])
+        return [_settings.mapSettingShowOnlineNotes get];
+    else if ([key isEqualToString:@"mapillary_layer"])
+        return _app.data.mapillary;
+    else if ([key isEqualToString:@"tracks"])
+        return _settings.mapSettingVisibleGpx.get.count > 0;
+    else if ([key isEqualToString:@"category_transport"])
+        return [_styleSettings isCategoryEnabled:@"transport"];
+    else if ([key isEqualToString:@"contour_lines_layer"])
+        return ![[_styleSettings getParameter:@"contourLines"].value isEqualToString:@"disabled"];
+    else if ([key isEqualToString:@"terrain_layer"])
+        return _app.data.terrainType != EOATerrainTypeDisabled;
+    else if ([key isEqualToString:@"overlay_layer"])
+        return _app.data.overlayMapSource != nil;
+    else if ([key isEqualToString:@"underlay_layer"])
+        return _app.data.underlayMapSource != nil;
+
+    if ([key hasPrefix:@"routes_"])
+    {
+        NSString *routesValue = _routesParameters[index].value;
+        return routesValue.length > 0 ? [key hasSuffix:HIKING_ROUTES_OSMC_ATTR] ? ![routesValue isEqualToString:@"disabled"] : [routesValue isEqualToString:@"true"] : NO;
+    }
+
+    return YES;
+}
+
+- (BOOL)isCollapsableGroup:(NSDictionary *)item
+{
+    return [item[@"type"] isEqualToString:NSStringFromClass([OATableCollapsableGroup class])];
+}
+
+- (NSDictionary *)getItem:(NSIndexPath *)indexPath
 {
-    NSDictionary *sectionData = tableData[section];
-    NSArray *cells = sectionData[@"cells"];
-    if (cells.count > 0)
+    NSArray *cells = tableData[indexPath.section][@"cells"];
+    if (cells.count > 0 && [self isCollapsableGroup:cells[0]])
     {
-        NSDictionary *cellData = cells[0];
-        if ([cellData[@"type"] isEqualToString:[OAAppModeCell getCellIdentifier]])
-            return 0.01;
-        else
-            return 34.0;
-        
+        OATableCollapsableGroup *group = cells[0][@"group"];
+        return !_isRoutesGroupOpen || indexPath.row == group.groupItems.count ? cells[0] : group.groupItems[indexPath.row];
     }
+
+    return cells[indexPath.row];
+}
+
+- (CGFloat)heightForHeader:(NSInteger)section
+{
+    NSArray *cells = tableData[section][@"cells"];
+    if (cells.count > 0)
+        return [cells[0][@"type"] isEqualToString:[OAAppModeCell getCellIdentifier]] ? 0.01 : 34.;
+
     return 0.01;
 }
 
-#pragma mark - OAAppModeCellDelegate
+- (CGFloat)heightForRow:(NSIndexPath *)indexPath estimated:(BOOL)estimated
```

Comment:
You do not need the heightForRow method for every tableview you create. In most cases, it is enough to set the appropriate values in the xib.